pydantic/config.py

```diff
@@ -96,7 +97,8 @@ def prepare_field(cls, field: 'ModelField') -> None:
         """
         Optional hook to check or modify fields during model creation.
         """
-        pass
+        if cls.partial:
```

Comment:
I think this could be a breaking/confusing change, since the default `prepare_field` does nothing, people will override it without calling `super()` and wonder why `partial` is not being respected.